tests/test_keys.py

```diff
@@ -197,6 +197,29 @@ def test_format_metadata_to_key(self):
 
 
 
+  def test_format_metadata_to_key_non_default_hash_algorithms(self):
```

Comment:
I think we should just roll these tests into `test_format_metadata_to_key()` – that way we keep all tests of this function together, as is the pattern throughout the rest of the tests.

Reply:
Addressed your comment by making changes and amending them to my last commit.